### PR TITLE
Features/log review

### PR DESCRIPTION
**Pull request by jmkerloch**
Review of `std::cout` use in Benders and BenderMpi so we can use this stream to display information to user. Previous `std::cout` was use in antares-xpansion python launcher as a log file with optimization results. Now we transfert `std::cout` to antares-xpansion python launcher `cout`

log file is not used anymore for end-to-end tests, we use out.jon result file.

std::cout content was changed as requested in #73 

Review of Benders TRACE option, result .json is always created.

### CODE REVIEW

File: src/tests/examples/example_test.py [hunk 1]
```diff
@@ -23,6 +20,15 @@ def find_log_path(output_dir):
     return op[0]
 
 
+def find_json_output(output_dir):
```
Comment (sgatto) [inline src/tests/examples/example_test.py:23]:
```suggestion
def get_first_json_filepath_output(output_dir):
```

File: src/tests/examples/example_test.py [hunk 2]
```diff
@@ -23,6 +20,15 @@ def find_log_path(output_dir):
...
+    op = []
+    for path in Path(output_dir).iterdir():
+        for log in Path(path / "lp").rglob('out.json'):
```
Comment (sgatto) [inline src/tests/examples/example_test.py:26]:
```suggestion
        for jsonpath in Path(path / "lp").rglob('out.json'):
```

File: src/tests/examples/example_test.py [hunk 3]
```diff
@@ -23,6 +20,15 @@ def find_log_path(output_dir):
     return op[0]
```
Comment (sgatto) [inline src/tests/examples/example_test.py:21]: is it still useful?

File: src/tests/examples/example_test.py [hunk 4]
```diff
@@ -36,102 +42,92 @@ def launch_xpansion(install_dir, study_path, method):
...
+               str(study_path), "--method", method, "--step", "full", "-n", "2"]
     process = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=None)
     output = process.communicate()
 
-    if process.returncode !=0 :
+    if process.returncode != 0:
         log_path = find_log_path(study_path / 'output')
-        #print log
+        # print log
         print(open(str(log_path), 'r').read())
```
Comment (sgatto) [inline src/tests/examples/example_test.py:54]: not needed in a test
can be removed

File: src/tests/examples/example_test.py [hunk 5]
```diff
@@ -36,102 +42,92 @@ def launch_xpansion(install_dir, study_path, method):
...
 
     # Check return value
     assert process.returncode == 0
 
 
-def check_investment_solution(study_path, expected_last_it,expected_investment_solution):
+def check_solution(study_path, expected_values, expected_investment_solution):
```
Comment (sgatto) [inline src/tests/examples/example_test.py:59]:
```suggestion
def verify_solution(study_path, expected_values, expected_investment_solution):
```

File: src/tests/examples/example_test.py [hunk 6]
```diff
@@ -36,102 +42,92 @@ def launch_xpansion(install_dir, study_path, method):
...
+    json_file = open(str(json_path), 'r')
+    json_data = json.load(json_file)
+
+    solution = json_data["solution"]
+    investment_solution = solution["values"]
+
+    np.testing.assert_allclose(solution["gap"], expected_values["gap"], rtol=1e-4, atol=0)
+    np.testing.assert_allclose(solution["investment_cost"], expected_values["investment_cost"], rtol=1e-4, atol=0)
+    np.testing.assert_allclose(solution["operational_cost"], expected_values["operational_cost"], rtol=1e-4, atol=0)
+    np.testing.assert_allclose(solution["overall_cost"], expected_values["overall_cost"], rtol=1e-4, atol=0)
```
Comment (sgatto) [inline src/tests/examples/example_test.py:72]: introduce constant
`RELATIVE_TOLERANCE=1e-4`

`atol=0` by default

```suggestion
    RELATIVE_TOLERANCE=1e-4
    np.testing.assert_allclose(solution["gap"], expected_values["gap"], rtol=RELATIVE_TOLERANCE)
    np.testing.assert_allclose(solution["investment_cost"], expected_values["investment_cost"], rtol=RELATIVE_TOLERANCE)
    np.testing.assert_allclose(solution["operational_cost"], expected_values["operational_cost"], rtol=RELATIVE_TOLERANCE)
    np.testing.assert_allclose(solution["overall_cost"], expected_values["overall_cost"], rtol=RELATIVE_TOLERANCE)
```

**Review by sgatto — APPROVED** [review]
I approve after considering the small refactoring suggestions

**Review by flomnes — COMMENTED** [review]
* `init_log` and `print_log` no longer write detailed logs. Is this intended ?

* 2 digits for candidate capacities as suggested by @MathildeFranconRTE

**Review by sgatto — APPROVED** [review]

**Pull request merged**